File: Backend/backend/src/main/java/com/example/backend/entity/Payout.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@Entity
@NoArgsConstructor
@Table(name = "payouts")
public class Payout {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tenant_id")
    private Tenant tenant;

    private Instant periodStart;
    private Instant periodEnd;

    private double grossRevenue;
    private double platformFeePercent;
    private double platformFee;
    private double netAmount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PayoutStatus status = PayoutStatus.PENDING;

    private Instant createdAt = Instant.now();
    private Instant paidAt;

    private String reference;

    @Column(columnDefinition = "TEXT")
    private String notes;

    public enum PayoutStatus {
        PENDING, PAID, ON_HOLD
    }
}

File: SuperAdmin/backend/SuperAdmin.API/Models/Payout.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperAdmin.API.Models;

// Maps the Spring-owned `payouts` table. status is stored as a string (Spring @Enumerated(STRING) +
// a DB CHECK constraint): PENDING | PAID | ON_HOLD — write exactly those values.
[Table("payouts")]
public class Payout
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("tenant_id")]
    public Guid? TenantId { get; set; }

    [Column("period_start")]
    public DateTime? PeriodStart { get; set; }

    [Column("period_end")]
    public DateTime? PeriodEnd { get; set; }

    [Column("gross_revenue")]
    public double GrossRevenue { get; set; }

    [Column("platform_fee_percent")]
    public double PlatformFeePercent { get; set; }

    [Column("platform_fee")]
    public double PlatformFee { get; set; }

    [Column("net_amount")]
    public double NetAmount { get; set; }

    [Column("status")]
    public string Status { get; set; } = "PENDING";

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("paid_at")]
    public DateTime? PaidAt { get; set; }

    [Column("reference")]
    public string? Reference { get; set; }

    [Column("notes")]
    public string? Notes { get; set; }
}
